src/main/java/org/springframework/cloud/deployer/spi/kubernetes/DefaultContainerFactory.java

```diff
@@ -273,9 +273,16 @@ private ProbeType getProbeType(ContainerConfiguration containerConfiguration) {
 	 */
 	List<String> createCommandArgs(AppDeploymentRequest request) {
 		List<String> cmdArgs = new LinkedList<>();
-		List<String> commandArgOptions = request.getCommandlineArguments().stream()
-				.map(arg-> arg.substring(0,arg.indexOf("=")).replaceAll("^--",""))
-				.collect(Collectors.toList());
+
+		List<String> commandArgOptions = new ArrayList<>();
+		for(String arg : request.getCommandlineArguments()) {
+			int indexOfAssignment = arg.indexOf("=");
+			if(indexOfAssignment < 0) {
+				commandArgOptions.add(arg);
```

Comment:
Should we also be scrubbing this option-only-variant for `--`?

Reply:
Its only stripped to make sure we are comparing like properties.   If I understand the question.

Reply:
I meant should we also make sure we strip the `--` from the `key-only` style args like we are doing w/ the `key=value` style args. The latest commit handles both and your tests cover all of the cases I can think of so I will resolve this now.